### DISCUSSION TITLE
Negative fleet numbers for lcomp

### DISCUSSION THREAD

**Discussion by kinmic** [ss3sim/ss3sim#426] [General]
Hi SS3 team,

I'm wondering if is it possible to set the fleet numbers for the length composition data as negative values, so expected values are calculated but length composition doesn't contribute to the overall model likelihood? I've tried changing the OM data file to have negative fleet numbers, but this doesn't seem to propagate through into the simulation. 

Is there a trick for doing this? Or any other trick for calculating estimated values for a data source but not including them in the overall model fit?

Thanks!

**Comment by kellijohnson-NOAA** [comment]
I am sorry but sss3sim is not set up to work with  negative fleets. If it were set up to work this way, you would need to do it within the dataframe that tells ss3sim how to sample though not within the dat file of the OM. Below is an example data frame that can be used to run a simulation
```
df <- ss3sim::setup_scenarios_defaults()
```
But, the parser that takes this data frame and tells ss3sim how to run does not know what to do with negative fleets. We could explore two other ways to  make this work, (1) run ss3sim_base rather than ss3sim_run or (2) run the scenario with positive fleets and then after the fact go in and change the dat file to have negative fleets. Let me know if you need help with either of those.

**Reply by kinmic** [reply]
Hi Kelli,

Ok, no problem. I tried both methods you suggested and found that running the sim, changing the data file in the em, and rerunning the model was the most efficient way.

Thanks for the help!
